grapher/controls/ControlsRow.tsx

```diff
@@ -0,0 +1,125 @@
+import React from "react"
+import { computed, action } from "mobx"
+import { observer } from "mobx-react"
+import {
+    Controls,
+    FilterSmallCountriesToggle,
+    HighlightToggle,
+    AbsRelToggle,
+    ZoomToggle,
+} from "./Controls"
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome"
+import { faPencilAlt } from "@fortawesome/free-solid-svg-icons/faPencilAlt"
+import { faPlus } from "@fortawesome/free-solid-svg-icons/faPlus"
+import { faExchangeAlt } from "@fortawesome/free-solid-svg-icons/faExchangeAlt"
+import { ScaleSelector } from "./ScaleSelector"
+
+@observer
+export class ControlsRow extends React.Component<{
+    controls: Controls
+}> {
+    static readonly height = 45
+
+    @action.bound onDataSelect() {
+        this.grapher.isSelectingData = true
+    }
+
+    @computed get grapher() {
+        return this.props.controls.props.grapher
+    }
+
+    @computed get controlsToRender() {
+        const grapher = this.grapher
+        const controls: JSX.Element[] = []
+
+        if (grapher.tab === "chart") {
+            const yAxis = grapher.activeTransform.yAxis
+            yAxis &&
+                yAxis.scaleTypeOptions.length > 1 &&
+                controls.push(
+                    <ScaleSelector scaleTypeConfig={yAxis} inline={true} />
+                )
+
+            grapher.canAddData &&
+                !grapher.hasFloatingAddButton &&
```

Comment:
Okay, we can probably get to the point where we just pass in these controls (or a props object with these flags). Not sure when that will happen but shouldn't be too hard to merge in.

Reply:
How about something like:
```
interface ChartType {
  ...
  controls: React.ReactElement[]
}

class LineChart extends ChartType {...}
class ScatterPlot extends ChartType {...}

class ControlsRow extends React.Component<> {
  render() {
     return <FancyUI items={grapher.activeChart.controls} />
     // or pass in as prop to avoid using grapher
  }
}
```

Reply:
Maybe even just using react children prop?